src/vs/editor/common/model/bracketPairColorizer/bracketPairColorizer.ts

```diff
@@ -82,19 +96,35 @@ export class BracketPairColorizer extends Disposable implements DecorationProvid
 		if (ownerId === undefined) {
 			return [];
 		}
+		if (!this.options.enabled) {
+			return [];
+		}
 		return this.cache.value?.object.getDecorationsInRange(range, ownerId, filterOutValidation) || [];
 	}
 
 	getAllDecorations(ownerId?: number, filterOutValidation?: boolean): IModelDecoration[] {
 		if (ownerId === undefined) {
 			return [];
 		}
+		if (!this.options.enabled) {
+			return [];
+		}
 		return this.cache.value?.object.getAllDecorations(ownerId, filterOutValidation) || [];
 	}
 
 	onDidChangeDecorations(listener: () => void): IDisposable {
 		return this.didChangeDecorationsEmitter.event(listener);
 	}
+
+	/**
+	 * Returns all bracket pairs that intersect the given range.
+	 * The result is sorted by the start position.
+	*/
+	getBracketPairsInRange(range: Range): BracketPair[] {
+		this.bracketsRequested = true;
```

Comment:
The tree is build the first time brackets are requested and stays alive until the model is disposed.